server/server.js

```diff
@@ -5,178 +5,195 @@ let https = require('https')
+                returnMessage = 'Thank you. If you like, you can reply with any further details about the incident.'
+            }
+            break
 
-    let installation = await db.getInstallationWithInstallationId(session.installationId)
+        case ALERT_STATE.WAITING_FOR_DETAILS:
+            returnMessage = 'Thank you. This session is now complete. (You don\'t need to respond to this message.)'
+            break
 
-    if (phoneNumber === installation.responderPhoneNumber) {
+        case ALERT_STATE.COMPLETED:
+            returnMessage = 'There is no active session for this button. (You don\'t need to respond to this message.)'
+            break
 
-        let stateMachine = new StateMachine(installation)
-        let {newSessionState, returnMessage} = stateMachine.processStateTransitionWithMessage(session, message)
-        await sendTwilioMessage(installation.responderPhoneNumber, session.phoneNumber, returnMessage);
-        await db.saveSession(newSessionState)
-        return 200;
-    } 
-    else {
-        log('Invalid Phone Number');
-        return 400;
+        default:
+            returnMessage = 'Thank you for responding. Unfortunately, we have encountered an error in our system and will deal with it shortly.'
+            break
     }
-}
 
-async function needToSendButtonPressMessageForSession(session) {
-    return (session.numPresses === 1 || session.numPresses === 3 || session.numPresses % 5 === 0)
+    return returnMessage
 }
 
-async function sendButtonPressMessageForSession(session) {
-
-    let installation = await db.getInstallationWithInstallationId(session.installationId)
+// Configure BraveAlerter
+const braveAlerter = new BraveAlerter(
+    getAlertSession,
+    getAlertSessionByPhoneNumber,
+    alertSessionChangedCallback,
+    true,
+    getReturnMessage,
+)
 
-    if (session.numPresses === 1) {
-        await sendTwilioMessage(installation.responderPhoneNumber, session.phoneNumber, 'There has been a request for help from Unit ' + session.unit.toString() + ' . Please respond "Ok" when you have followed up on the call.')
-        setTimeout(sendReminderMessageForSession, unrespondedSessionReminderTimeoutMillis, session.id)
-        setTimeout(sendStaffAlertForSession, unrespondedSessionAlertTimeoutMillis, session.id)
-    } 
-    else if (session.numPresses % 5 === 0 || session.numPresses === 3) {
-        await sendTwilioMessage(installation.responderPhoneNumber, session.phoneNumber, 'This in an urgent request. The button has been pressed ' + session.numPresses.toString() + ' times. Please respond "Ok" when you have followed up on the call.')
-    }
-}
-
-async function sendTwilioMessage(toPhoneNumber, fromPhoneNumber, message) {
-    try {
-        await client.messages.create({from: fromPhoneNumber, body: message, to: toPhoneNumber})
-            .then(message => log(message.sid))
-    }
-    catch(err) {
-        log(err)
-    }
-}
+// Add BraveAlerter's routes
+app.use(braveAlerter.getRouter())
```

Comment:
@schwarrrtz @mariocimet 
I remember you mentioning that DNS changes can sometimes take awhile. Do I need to do anything in the code to work around this change from `/message` to `/alert/sms`? Maybe have the one redirect to the other until the DNS change kicks in?

On the other hand, it looks like we can use this API call to change the SMS route for a given Twilio number: https://support.twilio.com/hc/en-us/articles/223136047-Configure-a-Twilio-Phone-Number-to-Receive-and-Respond-to-Messages#h_58646936-ad13-4940-aeb7-252b1d5aaa4e

I'm going to see if there's a way to get a list of all the Twilio numbers for an account.

Reply:
OK, I managed to do this programmatically. I added the script to `utils.js` and tested it against the Buttons Dev Twilio account. What do you think? Can we avoid the DNS change if we do this during deployment?

Reply:
Sounds like that script will be super useful during deployment!

The DNS changes we were talking about are at a higher level. It'll be something to consider when we change from `chatbot.brave.coop` to `buttons.brave.coop` or something like that. Other than updating the Twilio webhooks, we don't need to worry about changing the routes, as long as the FQDN stays the same

Reply:
If we want to wait on updating the Twilio webhooks (eg. so that deployment is simpler) we could add a redirect from `/message` to `/alert/sms` like you're suggesting. But that's not related to DNS stuff.